ClavaLaraApi/src-lara-clava/clava/clava/pass/SingleReturnFunction.js

```diff
@@ -0,0 +1,41 @@
+laraImport("lara.pass.Pass");
+laraImport("weaver.Query");
+
+class SingleReturnFunction extends Pass {
+  constructor() {
+    super("SingleReturnFunction");
+  }
+
+  _apply_impl($jp) {
+    if ($jp.joinpointType !== "function" || !$jp.isImplementation) {
+      return;
+    }
+    const $body = $jp.body;
+    const $returnStmts = Array.from(Query.searchFrom($body, "returnStmt"));
```

Comment:
Query.searchFrom($body, "returnStmt").get() will return an array with all the stmts. Is this what's intended?

Reply:
From my testing the query only returns all the return statements in this function body, and yes, they are all to be replaced.

Another question I have is if this pass should be applied to specific function instances, or if it should be applied to the program as a whole (perhaps being controlled by a clava data pragma flag).